### DISCUSSION TITLE
How to build msquic with dpdk on Linux platform

### DISCUSSION THREAD

**Discussion by 359547629** [microsoft/msquic#2645] [Q&A] [answered]
How to build msquic with dpdk on Linux platform?

**Comment by nibanks** [comment, marked as answer]
It isn't currently supported.

**Reply by thddaniel** [reply]
Hi [nibanks](https://github.com/nibanks), why the dpdk code was deleted？Is there a plan to support dpdk on  Linux platform? Thx
![image](https://user-images.githubusercontent.com/7098631/217136677-4c413069-99d6-4bce-9e89-2f7ef54e662a.png)

**Reply by nibanks** [reply]
Yes, eventually. We also support external contributions if you can't wait on us.

**Comment by dreamwind1985** [comment]
which performance is better, msquic on dpdk or on xdp?
I saw the performance of msquic on xdp is much better than the performance of msquic using network protocol stack of linux kernel. Does anybody know the performance of msquic on dpdk, and comparing with the performance on xdp, which is better?

**Reply by nibanks** [reply]
Since we don't currently have DPDK working, I can't tell you (for certain) what will eventually be the fastest. But I do suspect DPDK will eventually be the fastest; but I don't know by how much. It might be just a little bit faster. We haven't prioritized this work (DPDK support) because of the lack of committed customers.

**Reply by dreamwind1985** [reply]
Thanks very much for your response.
I did some performance test for gquic over dpdk， but I got only 5 percent improvement on dpdk comparing with the performance of gquic using linux kernel which I  set '/sys/module/kernel/parameters/uhash4_enable'  Y. I saw the most consuming was encryption and decryption functions. 
I used a user mode protocol stack for adapting gquic on dpdk, just like  https://github.com/Zhiyi-Zhang/f-stack-for-quic and https://github.com/F-Stack/f-stack. In this method, dpdk get the packet from netcard and notify read/write event to kernel.
Is there any better method for adapting quic on dpdk？

**Reply by nibanks** [reply]
It all depends on the metric you are using to measure performance and compare. For instance, latency is generally greatly improved by DPDK or XDP. Throughput might not be improved as much. As far as how to implement it all, there are lots of options. Generally, we try to implement as much as necessary ourselves to optimize performance.

**Reply by dreamwind1985** [reply]
I get it, thank you very much.
I will try to test the latency and throughput on dpdk and xdp.
